### PR TITLE
Add phi-deidentifier and date-annotator-stack (WIP)

### PR DESCRIPTION
**Pull request by tschaffter**
Implements https://github.com/nlpsandbox/nlpsandbox/issues/27

Issue 1 (date-annotator-stub): https://github.com/nlpsandbox/date-annotator-stub/issues/1
Issue 2 (phi-deidentifier): cannot access the app or api

### CODE REVIEW

**Comment by tschaffter** [comment]
@cascadianblue Can you have a look at Issue 2?

**Review by cascadianblue — COMMENTED** [review]
@tschaffter this is only an educated guess, but I think the problem is that the React app isn't set up to be [served from a subdirectory](https://create-react-app.dev/docs/deployment/#building-for-relative-paths).

I think I will have to have use build-time env vars like `REACT_APP_SUBDIRECTORY` and `REACT_APP_HOMEPAGE`. Is there a way to make a separate build with different env vars specifically for the AWS-hosted version?

**Comment by cascadianblue** [comment]
I think [this issue](https://github.com/nlpsandbox/phi-deidentifier-app/issues/22) will fix the problem

File: tools.nlpsandbox.io/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    restart: always
+    networks:
+      - nlp-sandbox-internal
+
+  # date-annotator-stub:
+  #   image: nlpsandbox/date-annotator-stub:1.0.0
+  #   container_name: date-annotator-stub
+  #   restart: always
+  #   networks:
+  #     - nlp-sandbox-internal
```
[the remark this answers is not in the document]
Comment (tschaffter) [reply tools.nlpsandbox.io/docker-compose.yml:16]: Yesterday I created the repo https://github.com/nlpsandbox/date-annotator-stub to show the output of the OpenAPI generated (server/) as is. I use this service as part of the documentation. This service is just a stub that returns "do some magic!" to all the endpoints of the Date Annotator.

File: tools.nlpsandbox.io/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      - ./nginx/nginx.conf:/etc/nginx/nginx.conf:ro
+      - ./nginx/templates:/etc/nginx/templates:ro
+    networks:
+      - nlp-sandbox
+      - nlp-sandbox-internal
+    ports:
+      - "80:80"
+    depends_on:
+      - phi-deidentifier-app
+      # - date-annotator-stub
```
[the remark this answers is not in the document]
Comment (tschaffter) [reply tools.nlpsandbox.io/docker-compose.yml:84]: For now ignore the date-annotator-stub (see above). The docker image published on Docker Hub has a problem which I still have to resolve.

**Comment by tschaffter** [comment]
@thomasyu888 I decided to no longer deploy multiple services like the PHI Deidentifier and the Date Annotator Stub on one EC2 for the following reasons:

- We need to customize the service so that the base path can be specified as part of the configuration (e.g. using environment variables). However, I don't want to do that for the Date Annotator Stub because I want to keep it as the OpenAPI generator creates it. I also don't want to increase the complexity by creating another Docker image for this "stub" service.
- Most importantly, we already know how to deploy one service or stack of services on an EC2. You do it for the Data Node. It is also more scalable to isolate services, for example the date annotator stub has nothing to do with the PHI Deidentifier. Also if we want to deploy another service in the future, we don't have to stop and restart the existing service. More scalable thus and less complex because we don't have to maintain the extra docker-compose.yml I was working on.

I'm closing this PR now and will open new tickets to track the deployment of the PHI Deidentifier other services.

**Pull request closed without merging**